contracts/RegisterDomain.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+contract RegisterDomain {
+
+    struct Domain {
+        address owner;
+    }
+
+    mapping(string => Domain) public domains;
+    uint256 public immutable registrationDeposit = 1 ether;
+
+    event DomainRegistered(string domain, address owner, uint256 amount);
+    event DomainReleased(string domain, address owner, uint256 amount);
+
+    modifier isValidDomain(string memory _domain) {
+        require(!containsDot(_domain), "Only top-level domains are allowed");
+        _;
+    }
+
+    modifier isAvailable(string memory _domain) {
+        require(domains[_domain].owner == address(0), "Domain registered");
```

Comment:
Better to rephrase it to something like "Domain already exists!". Otherwise error message may be not clear to the reader.